dev/tools/push-images

```diff
@@ -83,81 +88,61 @@ def build_and_push_image(args, service_name, srcdir, dockerfile_path):
-    repo_root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
-    examples_dir = os.path.join(repo_root, "examples")
-    if not os.path.isdir(examples_dir):
-        return
+    images_json_path = os.path.join(repo_root, "dev/tools/shared/images.json")
+    if not os.path.isfile(images_json_path):
+        print(f"Error: images.json not found at {images_json_path}")
 
-    for example in os.listdir(examples_dir):
-        example_dir = os.path.join(examples_dir, example)
-        if not os.path.isdir(example_dir):
-            continue
+    with open(images_json_path, "r") as f:
+        images_to_build = json.load(f)
 
-        dockerfile_path = os.path.join(example_dir, "Dockerfile")
-        if not os.path.isfile(dockerfile_path):
-            continue
+    create_buildx_builder_if_not_exists()
 
-        service_name = example
+    for image_info in images_to_build:
+        service_name = image_info["service_name"]
+        context_dir = image_info["context_dir"]
+        dockerfile_path = image_info["dockerfile"]
 
-        print(f"create Docker buildx builder for {service_name}")
-        create_buildx_builder_if_not_exists()
-        print(f"building image for {service_name}")
-        build_and_push_image(args, service_name, example_dir, "Dockerfile")
+        build_context_abs = os.path.join(repo_root, context_dir)
+        dockerfile_path_abs = os.path.join(repo_root, dockerfile_path)
 
+        print(
+            f"building image for {service_name} from {dockerfile_path_abs} in {build_context_abs}"
+        )
+        build_and_push_image(args, service_name, build_context_abs, dockerfile_path_abs)
 
-def main(args):
-    """Builds and pushes docker images."""
-    golang.for_each_module(lambda mod_dir: push_image_for_go_mod(args, mod_dir))
-    push_images_for_examples(args)
-    # TODO: Handle image building and pushing logic better
-    
 
 if __name__ == "__main__":
     # parse arguments and call
     parser = argparse.ArgumentParser(description="Build and push Docker images")
-    parser.add_argument("--image-prefix",
-                        dest="image_prefix",
-                        help="Prefix for the image name. requires slash at the end if a path",
-                        type=str,
-                        default=os.getenv("IMAGE_PREFIX"))
-    parser.add_argument("--extra-image-tag",
-                        dest="extra_image_tags",
-                        help="Additional tags to use when publishing images.",
-                        type=str,
-                        action="append",
-                        default=[])
-    parser.add_argument("--kind-cluster-name",
-                        dest="kind_cluster_name",
-                        help="Name of the kind cluster. If set, images will be loaded into the cluster",
-                        type=str,
-                        default=os.getenv("KIND_CLUSTER_NAME"))
+    parser.add_argument(
```

Comment:
This change (L124-145) doesn't seem necessary. Please revert it to keep the git history clean.